mteb/models/jina_models.py

```diff
@@ -185,6 +191,139 @@ def encode(
         return embeddings
 
 
+class JinaV4Wrapper(JinaWrapper):
+    """following the hf model card documentation."""
+
+    jina_task_to_prompt = {
+        "retrieval.query": "Query: ",
+        "retrieval.passage": "Passage: ",
+        "text-matching": "Query: ",
+    }
+
+    def __init__(
+        self,
+        model: str,
+        revision: str | None = None,
+        model_prompts: dict[str, str] | None = None,
+        **kwargs,
+    ) -> None:
+        requires_package(self, "peft", model, "pip install 'mteb[peft]'")
+        import peft  # noqa: F401
+
+        super().__init__(
+            model=model,
+            revision=revision,
+            model_prompts=model_prompts,
+            **kwargs,
+        )
+
+    def encode(
+        self,
+        sentences: Sequence[str],
+        *,
+        task_name: str,
+        prompt_type: PromptType | None = None,
+        **kwargs: Any,
+    ) -> np.ndarray:
+        prompt_name = self.get_prompt_name(self.model_prompts, task_name, prompt_type)
+        if prompt_name:
+            logger.info(
+                f"Using prompt_name={prompt_name} for task={task_name} prompt_type={prompt_type}"
+            )
+        else:
+            logger.info(
+                f"No model prompts found for task={task_name} prompt_type={prompt_type}"
+            )
+        logger.info(f"Encoding {len(sentences)} sentences.")
+
+        # Get Jina-specific parameters
+        jina_task_name = self.model_prompts.get(prompt_name) if prompt_name else None
+        jina_prompt = (
+            self.jina_task_to_prompt.get(jina_task_name) if jina_task_name else None
+        )
+
+        # Override task for programming-related content
+        jina_task_name = get_programming_task_override(task_name, jina_task_name)
+
+        embeddings = self.model.encode(
+            sentences,
+            task=jina_task_name.split(".")[0] if jina_task_name else None,
```

Comment:
I think your should pass query or passage, not `retrieval`, but this will break with `text-matching`
```suggestion
            task=jina_task_name.split(".")[1] if jina_task_name else None,
```

Reply:
Our model expects `retrieval`, `code`, or `text-matching` as the task. So, the current implementation should work without any issue. I have also tested it